src/Bicep.LangServer/Server.cs

```diff
@@ -99,6 +100,10 @@ public async Task RunAsync(CancellationToken cancellationToken)
 
             server.LogInfo($"Running on processId {Environment.ProcessId}");
 
+            var mcrCompletionProvider = server.GetRequiredService<IModulesMetadataProvider>();
+            var mcrModulesMetadatInitializationResult = await mcrCompletionProvider.Initialize();
+            server.LogInfo(mcrModulesMetadatInitializationResult);
```

Comment:
We should avoid blocking server startup on external I/O. Happy to chat through alternative options if you like!

Reply:
I moved the logic to ModulesMetadataProvider here:
https://github.com/Azure/bicep/pull/9913/commits/12739e5deb1e771a4d3c3c51063acf8cf0f0980b

Could you please take a look and let me know if you have any concerns?

Thanks!

Reply:
Thanks! I've posted a more detailed comment below - this is definitely an improvement, but I still think there are other scenarios we may want to consider.

Marking as resolved as you've addressed my feedback here.